### ISSUE TITLE
Compile error 

### ISSUE THREAD

**Issue by zpliu1126** [baoxingsong/dCNS#15]
Hi~ baoxingsong

I compile the dCNS with this pipline

```bash
#! **CentOS Linux release 7.5.1804 (Core)**
cd dCNS-0.3
module load GCC/6.2.0-2.27
module load CMake/3.16.5 
cmake -DCMAKE_CXX_COMPILER=$(which g++) -DCMAKE_C_COMPILER=$(which gcc) CMakeLists.txt
make
```

compile error
```bash
software/dCNS-0.3/test/impl/findSimilarFragmentsForPairedSequenceControlLayer4.cpp:241:119: error: no matching function for call to ‘cut1Gap(std::__cxx11::string&, std::__cxx11::string&, std::__cxx11::string&, int32_t&, int32_t&, int32_t&, int32_t&, int32_t&, int32_t&, int32_t&, int32_t&, int32_t&, int32_t&, bool&, double&, double&, int32_t&, int32_t&, std::__cxx11::string&, std::__cxx11::string&, double&)’
             step_size, matrix_boundary_distance, onlySyntenic, lambda, kValue, w, xDrop, refFasta, queryFasta, pvalues);
                                                                                                                       ^
In file included from /data/cotton/zhenpingliu/LZP_fiberFullPopulationRNAseq/software/dCNS-0.3/test/impl/findSimilarFragmentsForPairedSequenceControlLayer4.cpp:14:0:
make[2]: *** [CMakeFiles/dCNS.dir/test/impl/findSimilarFragmentsForPairedSequenceControlLayer4.cpp.o] Error 1
make[1]: *** [CMakeFiles/dCNS.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by baoxingsong** [comment]
Sorry, I could not figure out the problem from your above information. Could you share all the outputs from cmake and make, please?

**Comment by zpliu1126** [comment]
thank you for you response.
I compile it successfully by docke in my window machine. But failed in the Centos system. I guess it may be I'm not a administrator for centos. Can you provide a copy of the version information of the software in the docker environment, or a docker image.

**Comment by baoxingsong** [comment]
I am not an expert on docker. But you could load the latest GCC module on your cluster, and try again.

After changing to newer GCC, please make sure you deleted the dCNS folder completely, and clone the source code again.

**Comment by conniecl** [comment]
Hi @zpliu1126 @baoxingsong ，
I’m wondering if that problem solved yet？ I have met the same "Compile error" in my "Ubuntu 20.04.6" system, but without in the "Ubuntu 18.04.6". After checking the difference of gcc and cmake in that two machine, and then change the version for 20.04.6, the bug still here. Sincerely hope can get your suggestions.
[make.log](https://github.com/baoxingsong/dCNS/files/13520670/make.log)

```
/mnt/sda/lchen/software/dCNS-0.3/test/impl/findSimilarFragmentsForPairedSequenceControlLayer4.cpp:241:119: error: no matching function for call to 'cut1Gap(std::__cxx11::string&, std::__cxx11::string&, std::__cxx11::string&, int32_t&, int32_t&, int32_t&, int32_t&, int32_t&, int32_t&, int32_t&, int32_t&, int32_t&, int32_t&, bool&, double&, double&, int32_t&, int32_t&, std::__cxx11::string&, std::__cxx11::string&, double&)'
             step_size, matrix_boundary_distance, onlySyntenic, lambda, kValue, w, xDrop, refFasta, queryFasta, pvalues);
                                                                                                                       ^
In file included from /mnt/sda/lchen/software/dCNS-0.3/test/impl/findSimilarFragmentsForPairedSequenceControlLayer4.cpp:14:0:
/mnt/sda/lchen/software/dCNS-0.3/test/impl/../../controlLayer.h:91:6: note: candidate: void cut1Gap(std::__cxx11::string&, std::__cxx11::string&, std::__cxx11::string&, int&, int&, int&, int&, int32_t&, int32_t&, int32_t&, int&, bool&, const double&, const double&, int&, const int&, const string&, const string&, double&)
 void cut1Gap(std::string & _input,  std::string & _reference, std::string & _output,
      ^~~~~~~
/mnt/sda/lchen/software/dCNS-0.3/test/impl/../../controlLayer.h:91:6: note:   candidate expects 19 arguments, 21 provided
/mnt/sda/lchen/software/dCNS-0.3/test/impl/../../controlLayer.h:109:5: note: candidate: int cut1Gap(int, char**)
 int cut1Gap(int argc, char** argv);
     ^~~~~~~
/mnt/sda/lchen/software/dCNS-0.3/test/impl/../../controlLayer.h:109:5: note:   candidate expects 2 arguments, 21 provided
/mnt/sda/lchen/software/dCNS-0.3/test/impl/findSimilarFragmentsForPairedSequenceControlLayer4.cpp:219:13: warning: unused variable 'zDrop' [-Wunused-variable]
     int32_t zDrop = 50; // this one should be slightly larger than openGapPenalty2
             ^~~~~
/mnt/sda/lchen/software/dCNS-0.3/test/impl/findSimilarFragmentsForPairedSequenceControlLayer4.cpp:220:13: warning: unused variable 'bandwidth' [-Wunused-variable]
     int32_t bandwidth = 20000; // using very large value, 12000 there are some strange result, should be checked very carefully
             ^~~~~~~~~
make[2]: *** [CMakeFiles/dCNS.dir/build.make:399: CMakeFiles/dCNS.dir/test/impl/findSimilarFragmentsForPairedSequenceControlLayer4.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:69: CMakeFiles/dCNS.dir/all] Error 2
make: *** [Makefile:84: all] Error 2

```

**Comment by conniecl** [comment]
Leave the messages here to avoid someone met the same compile error (since I met and checked all the version for gcc, cmake, but found it caused by the dCNS version). This error was made by the version we used, please use the master version (v0.4) instead of v0.3, you would get this wonderful tool installed well.
